nmigen/lib/fifo.py

```diff
@@ -380,8 +380,8 @@ def elaborate(self, platform):
             r_empty.eq(consume_r_gry == produce_r_gry),
         ]
 
-        m.d[self._w_domain] += self.w_level.eq((produce_w_bin - consume_w_bin)[:self._ctr_bits-1])
-        m.d.comb += self.r_level.eq((produce_r_bin - consume_r_bin)[:self._ctr_bits-1])
+        m.d[self._w_domain] += self.w_level.eq((produce_w_bin - consume_w_bin))
+        m.d.comb += self.r_level.eq((produce_r_bin - consume_r_bin))
```

Comment:
Why did you change this? I thought this PR only affects AsyncFIFOBuffered.

Reply:
If I remember correctly this is needed so that the counter doesnt overflow (as tested by the testsuite)

Reply:
should I split that change into its own pr then?

Reply:
I'm still not entirely certain why it's done; is this basically fixing a second bug unrelated to the `w_level` issue?

Reply:
without that change https://github.com/nmigen/nmigen/pull/512/files#diff-5a301e8309610ecfa06031693c54b1c02c7d05a74fa97224ecffd0470d950183R339 would fail as the reported level would be 0

Reply:
Yes, please split it out.